common/src/main/java/io/pravega/common/util/BufferView.java

```diff
@@ -142,14 +154,43 @@ default void release() {
      */
     static BufferView wrap(List<BufferView> components) {
         if (components.size() == 0) {
-            return new ByteArraySegment(new byte[0]);
+            return empty();
         } else if (components.size() == 1) {
             return components.get(0).slice();
         } else {
             return new CompositeBufferView(components);
         }
     }
 
+    /**
+     * Creates a new {@link BufferViewBuilder} that can be used to construct composite {@link BufferView} instances.
+     *
+     * @return A new {@link BufferViewBuilder} with default initial component count.
+     */
+    static BufferViewBuilder builder() {
```

Comment:
From the abstract methods in this class in which this applies, would it make sense to clearly state in the javadoc what of them are expected to be zero-copy and which of them are not? I assume that you have an idea in mind for this, so perhaps this can be written as a guideline for future implementations of this interface. The thing is that, when working with `ByteBuffer` and `ByteBuf`, we sometimes have to go to the implementation to see whether an actual copy of the buffer is made or not, which is confusing. It would be great to use in this family of classes (this one and the ones implementing it) an annotation like `@ZeroCopy` or similar to make users rapidly aware of the costs of the different operations involved in this class.

Reply:
Some of them are self-explanatory (the ones containing `copy` in their name). I updated the Javadoc for some of the others to make it clear. I would prefer not to go down the annotation path as it would be something very specific which only applies to this class.